splice_machine/src/main/java/com/splicemachine/backup/BackupSystemProcedures.java

```diff
@@ -657,4 +656,41 @@ public static void SYSCS_RESTORE_SCHEMA(String destSchema,
         }
     }
 
+    public static void SYSCS_ROLLBACK_DATABASE_TO_TRANSACTION(long transactionId,
+                                                              ResultSet[] resultSets) throws StandardException, SQLException {
+        IteratorNoPutResultSet inprs = null;
+
+        Connection conn = SpliceAdmin.getDefaultConn();
+        LanguageConnectionContext lcc = conn.unwrap(EmbedConnection.class).getLanguageConnection();
+        try {
+            TransactionController tc=lcc.getTransactionExecute();
+            tc.elevate("rollback");
+            BackupManager backupManager = EngineDriver.driver().manager().getBackupManager();
+
+            long currentTxId = lcc.getTransactionExecute().getActiveStateTxId();
+
+            // Set Restore Mode to prevent other workloads from running
+            DDLMessage.DDLChange change = ProtoUtil.createRestoreMode(currentTxId);
+            String changeId = DDLUtils.notifyMetadataChange(change);
+            
+            // Rollback
+            backupManager.rollbackDatabase(transactionId, currentTxId);
+            
+            // Finish Restore Mode
+            DDLUtils.finishMetadataChange(changeId);
+
+            // Leave restore mode to allow other workload
+            change = ProtoUtil.createLeaveRestoreMode(currentTxId);
+            changeId = DDLUtils.notifyMetadataChange(change);
+            DDLUtils.finishMetadataChange(changeId);
+
+            EngineDriver.driver().manager().getUpgradeManager().restartOlapServer();
+
+            SpliceAdmin.INVALIDATE_GLOBAL_DICTIONARY_CACHE();
+        } catch (Throwable t) {
+            resultSets[0] = ProcedureUtils.generateResult("Error", t.getLocalizedMessage());
+            SpliceLogUtils.error(LOG, "Database rollback error", t);
+            t.printStackTrace();
+        }
+    }
 }
```

Comment:
What if there are running txns before database rollback, do we want to wait for them to complete, or ignore them and let them eventually fail?

Reply:
Wrote Jira for this issue: https://splicemachine.atlassian.net/browse/DB-11332